### PR TITLE
Retrieving top and recent stories from Designer News V2 API

### PR DESCRIPTION
**Pull request by florina-muntenescu**

### CODE REVIEW

File: app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt [hunk 1]
```diff
@@ -34,7 +35,9 @@ class DesignerNewsRepository(
     private val inflight: MutableMap<String, Call<*>> = HashMap()
 
     fun loadTopStories(page: Int, callback: LoadSourceCallback) {
-        val topStories = service.getTopStories(page)
+        val topStories = if (BuildConfig.DESIGNER_NEWS_V2)
+            service.getTopStoriesV2(page) else
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt]: add braces

File: app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt [hunk 2]
```diff
@@ -54,7 +57,9 @@ class DesignerNewsRepository(
     }
 
     fun loadRecent(page: Int, callback: LoadSourceCallback) {
-        val recentStoriesCall = service.getRecentStories(page)
+        val recentStoriesCall = if (BuildConfig.DESIGNER_NEWS_V2)
+            service.getRecentStoriesV2(page) else
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/data/api/designernews/DesignerNewsRepository.kt]: ditto

File: app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java [hunk 3]
```diff
@@ -31,18 +32,22 @@
...
     public final int comment_count;
     public final int vote_count;
-    public final Date created_at;
     public final long user_id;
-    public final String user_display_name;
-    public final String user_portrait_url;
+    public final Date created_at;
+    public final StoryLinks links;
+    @Nullable
+    public final String user_display_name;   // Gone
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java]: nit: fix whitespace here and below before `//`
Also, where have the values gone to?

File: app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java [hunk 4]
```diff
@@ -95,6 +102,7 @@ protected Story(Parcel in) {
         } else {
             comments = null;
         }
+        this.links = in.readParcelable(StoryLinks.class.getClassLoader());
```
Comment (keyboardsurfer) [inline app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java]: `this.` not needed in this constructor.

File: app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java (hunk 3 above)
Comment (florina-muntenescu) [reply app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java]: they were removed in v2. :( 
Keeping them for now in the Story, so we don't start refactoring the world in one PR

Comment (keyboardsurfer) [reply app/src/main/java/io/plaidapp/data/api/designernews/model/Story.java]: Makes sense. In this case `removed in V2` would be clearer, right?

**Pull request merged**